src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1662,8 +1660,6 @@ internal static uint GetRefEscape(BoundExpression expr, uint scopeOfTheContainin
                         scopeOfTheContainingExpression);
             }
 
-            //PROTOTYPE(span): when all relevant cases are handles, consider making the switch above comprehensive and Assert or throw Unreachable on unhandled node kinds.
```

Comment:
Why did we give up on this?

Reply:
Because the remaining cases are rvalues, and there are too many ways of producing rvalues, and we will add new ways to create them in the future but we don't want to keep adding to this switch.

---
In reply to: [138700246](https://github.com/dotnet/roslyn/pull/21982#discussion_r138700246) [](ancestors = 138700246)

Reply:
Is there any way we can detect that new nodes should be handled here (and corresponding switch below)? Maybe we add a test with list of handled kinds? Or an assert here with kinds that we intentionally let through?

---
In reply to: [138711134](https://github.com/dotnet/roslyn/pull/21982#discussion_r138711134) [](ancestors = 138711134,138700246)

Reply:
I did not give up on this. I think once all nodes that can come through here are handle we should add `case default` with an assert. There is a bug tracking this.

---
In reply to: [138729843](https://github.com/dotnet/roslyn/pull/21982#discussion_r138729843) [](ancestors = 138729843,138711134,138700246)

Reply:
Neal is right though defaulting to "RValue" is mostly ok, since that is the vast majority of expressions.

---
In reply to: [138748410](https://github.com/dotnet/roslyn/pull/21982#discussion_r138748410) [](ancestors = 138748410,138729843,138711134,138700246)

Reply:
I will consider this resolved, since it is tracked by [https://github.com/dotnet/roslyn/issues/22077](https://github.com/dotnet/roslyn/issues/22077)

---
In reply to: [138748431](https://github.com/dotnet/roslyn/pull/21982#discussion_r138748431) [](ancestors = 138748431,138748410,138729843,138711134,138700246)